server/modules/category/manager/categoryManager.js

```diff
@@ -151,26 +151,43 @@ const updateLevelProp = async (user, surveyId, categoryUuid, levelUuid, key, val
     }
   })
 
-const updateItemProp = async (user, surveyId, categoryUuid, itemUuid, key, value, client = db) =>
+const _newCategoryItemUpdateLogActivity = (categoryUuid, item, key, value, system) => ActivityLog.newActivity(
+  ActivityLog.type.categoryItemPropUpdate,
+  {
+    [ActivityLog.keysContent.uuid]: CategoryItem.getUuid(item),
+    [ActivityLog.keysContent.code]: CategoryItem.getCode(item),
+    [ActivityLog.keysContent.categoryUuid]: categoryUuid,
+    [ActivityLog.keysContent.levelUuid]: CategoryItem.getLevelUuid(item),
+    [ActivityLog.keysContent.key]: key,
+    [ActivityLog.keysContent.value]: value
+  },
+  system
+)
+
+const updateItemProp = async (user, surveyId, categoryUuid, itemUuid, key, value, validateCategory = true, client = db) =>
   await client.tx(async t => {
     const item = await CategoryRepository.updateItemProp(surveyId, itemUuid, key, value, t)
     await Promise.all([
       markSurveyDraft(surveyId, t),
-      ActivityLogRepository.insert(user, surveyId, ActivityLog.type.categoryItemPropUpdate, {
-        [ActivityLog.keysContent.uuid]: itemUuid,
-        [ActivityLog.keysContent.code]: CategoryItem.getCode(item),
-        [ActivityLog.keysContent.categoryUuid]: categoryUuid,
-        [ActivityLog.keysContent.levelUuid]: CategoryItem.getLevelUuid(item),
-        [ActivityLog.keysContent.key]: key,
-        [ActivityLog.keysContent.value]: value
-      }, false, t)
+      ActivityLogRepository.insertMany(user, surveyId, [_newCategoryItemUpdateLogActivity(categoryUuid, item, key, value, false)], t)
```

Comment:
why insertMany if you are inserting only one activityLog?

Reply:
It's there only to reuse the generation of the activity log with _newCategoryItemUpdateLogActivity, maybe we can have another function in ActivityLogRepository for it

Reply:
I'm not sure I understand; the method _newCategoryItemUpdateLogActivity returns one activity log. then you pass it within an array at line 172. Am I missing something?